docker/convert.go

```diff
@@ -114,7 +117,7 @@ func ports(c *config.ServiceConfig) (map[nat.Port]struct{}, nat.PortMap, error)
 }
 
 // Convert converts a service configuration to an docker API structures (Config and HostConfig)
-func Convert(c *config.ServiceConfig, ctx project.Context) (*container.Config, *container.HostConfig, error) {
+func Convert(c *config.ServiceConfig, ctx project.Context, clientFactory composeclient.Factory) (*container.Config, *container.HostConfig, error) {
```

Comment:
Yeah, I do agree, having `client.Factory` here is a pain. I think we can refactor some code to get this out of here but this will be easier with the `Container` refactoring :angel:.

I like the idea of making the conversion process in a separate package :wink:.